### PR TITLE
Made turret point directly towards mouse

### LINKED ISSUE #1: [BUG]: Rotations are not rendered properly
### Description

(Submitting as the head developer, because I'm not sure how long this bug will persists and other users might submit it too)

Node rotations are not accurate to their `rotation` property.
`utils.math.rotationTo()` and `utils.math.rotateDirection()` work properly.

Here the line is rendered from the turret's position to the mouse position:
image
As you can see, the bullets follow the line and move towards the mouse, but their rotation is not straight.
image

### Steps to reproduce

Move your mouse around the turret and observe its rotation. Fire bullets and observe their movement and texture rotation. 

### Game version

*

### Platform

Linux

### PR DESCRIPTION
**Pull request by gera2342**
I adjust mouse coords by world aspect ratio so they are not stretched (like if you would set `data.width` equal to `data.height`). Idk how you would want to expose `data.width` and `data.height` to your Turret class, so I just made `data` global. You might as well precompute world aspect ratio since `data.width` and `data.height` are constant.

Fixes #1

Also you may want to make bullets come out of turret's muzzle, not its center, since they are still misaligned during rotation.

### CODE REVIEW

**Comment by Wolfyxon** [comment]
Well the issue doesn't exactly lie within the turret but all nodes

File: src/main.lua [hunk 1]
```diff
@@ -1,4 +1,4 @@
-local data = require("data")
```
Comment (Wolfyxon) [inline src/main.lua:1]: `data` should not be global but imported where needed.

File: src/scenes/game/turret/turret.lua [hunk 2]
```diff
@@ -70,7 +72,15 @@ end
 function Turret:update(delta)
     if (love.mouse and love.mouse.isCursorSupported()) or ((not love.mouse or not love.mouse.isCursorSupported()) and utils.system.isMousePressed()) then
         local x, y = utils.system.getMousePos()
-        self.targetRotation = self:rotationTo(x, y) 
+        self.bulletTargetRotation = self:rotationTo(x, y)
+        x = x - self.x
+        y = y - self.y
+        if data.width > data.height then
+            x = x*(data.width/data.height)
```
Comment (Wolfyxon) [inline src/scenes/game/turret/turret.lua:79]: Please use spaces in calculations to make them more readable, for example
```
a + b * c
```

**Review by Wolfyxon — CHANGES_REQUESTED** [review]

**Comment by jiffygist** [comment]
did the changes

**Comment by Wolfyxon** [comment]
Merging as a temporary fix

**Pull request merged**